src/nvim/path.c

```diff
@@ -597,11 +615,11 @@ static size_t do_path_expand(garray_T *gap, const char_u *path,
   *s = NUL;
 
   Directory dir;
-  // open the directory for scanning
-  if (os_scandir(&dir, *buf == NUL ? "." : (char *)buf)) {
+  if (os_scandir(&dir, *buf == NUL ? "." : (char *)buf)
+      || os_isdir(*buf == NUL ? (char_u *)"." : (char_u *)buf)) {
     // Find all matching entries.
     char_u *name;
-    while((name = (char_u *) os_scandir_next(&dir))) {
+    while((name = (char_u *) os_scandir_next_with_dots(&dir)) && name != NULL) {
```

Comment:
Before this first call, should initialize the internal state with this call:

    os_scandir_next_with_dots(NULL);